pico_w/mqtt/mqtt_client/mqtt_client.c

```diff
@@ -0,0 +1,161 @@
+//
+// Created by elliot on 25/05/24.
+//
+#include "pico/stdlib.h"
+#include "hardware/gpio.h"
+#include "hardware/irq.h"
+#include "hardware/adc.h"
+#include "pico/cyw43_arch.h"
+#include "lwip/apps/mqtt.h"
+
+
+// Temperature
+#define TEMPERATURE_UNITS 'C'
+
+// WiFi
+#define WIFI_SSID "YOUR_WIFI_SSID"
+#define WIFI_PASSWORD "YOUR_WIFI_PASSWD"
```

Comment:
Rather than explicitly defining these here, it might be better to just pick them up "automatically" from the CMake settings? See https://github.com/raspberrypi/pico-examples/blob/master/pico_w/wifi/ntp_client/picow_ntp_client.c and https://github.com/raspberrypi/pico-examples/blob/master/pico_w/wifi/CMakeLists.txt and section 2.2 of https://datasheets.raspberrypi.com/picow/connecting-to-the-internet-with-pico-w.pdf